File: answers/other/coding/Maximum Subarray.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        
        sum = 0
        max_sum = -sys.maxint - 1
        for i in nums:
            if sum < 0:
                sum = 0
            sum += i
            max_sum = max(max_sum, sum)
        return max_sum

File: answers/leetcode/Maximum Subarray/Maximum Subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        if (null == nums || nums.length == 0) {
            return 0;
        }
        
        int max = nums[0];
        int current = nums[0];
        for (int i = 1; i < nums.length; i ++) {
            current = Math.max(current + nums[i], nums[i]);
            max = Math.max(current, max);
        }
        return max;
    }
}